File: app/src/main/java/org/favcode54/BareWebActivity.java
package org.favcode54;

import android.annotation.SuppressLint;
import android.os.Build.VERSION;
import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.view.View;
import android.webkit.WebSettings;
import android.webkit.WebSettings.LayoutAlgorithm;
import android.webkit.WebView;
import android.widget.ProgressBar;

import com.android.volley.DefaultRetryPolicy;
import com.android.volley.NetworkResponse;
import com.android.volley.Response;
import com.android.volley.toolbox.StringRequest;

import org.favcode54.R.id;
import org.favcode54.network.MyVolleyError;
import org.favcode54.network.VolleyCache;
import org.favcode54.network.VolleySingleton;
import org.jetbrains.annotations.NotNull;

import java.util.concurrent.TimeUnit;

import kotlin.jvm.internal.Intrinsics;

public final class BareWebActivity extends BaseActivity {
    public static final String URL = "PAGE_URL";
    public static final String TITLE = "PAGE_TITLE";

    private ProgressBar progressBar;
    private WebView webView;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.activity_bare_web);
        this.setSupportActionBar(findViewById(id.toolbar));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setTitle(getIntent().getStringExtra(TITLE));

        applyFont(findViewById(id.rootView));
        progressBar = findViewById(id.progressBar);
        webView = findViewById(id.webView);
        tintProgressBar(progressBar);
        String url = getIntent().getStringExtra(URL);
        initWebView();
        loadPageHtml(url);
    }

    @SuppressLint({"SetJavaScriptEnabled", "AddJavascriptInterface"})
    private final void initWebView() {
        WebSettings webSettings = webView.getSettings();
        Intrinsics.checkExpressionValueIsNotNull(webSettings, "webSettings");
        webSettings.setJavaScriptEnabled(true);
        webView.setWebViewClient(new MyWebViewClient(this));
        if (VERSION.SDK_INT >= 19) {
            webSettings.setLayoutAlgorithm(LayoutAlgorithm.TEXT_AUTOSIZING);
        } else {
            webSettings.setLayoutAlgorithm(LayoutAlgorithm.NORMAL);
        }

    }

    private void loadPageHtml(final String url) {
        progressBar.setVisibility(View.VISIBLE);
        StringRequest request = new StringRequest(url, response -> {

            progressBar.setVisibility(View.GONE);
            webView.loadDataWithBaseURL(url, response, "text/html", "UTF-8", null);

        }, error -> {
            String error1 = MyVolleyError.errorMessage(error, BareWebActivity.this);
            Snackbar.make(findViewById(id.rootView), error1, Snackbar.LENGTH_LONG).show();
            BareWebActivity.this.finish();
        }) {
            @NotNull
            protected Response<String> parseNetworkResponse(NetworkResponse response) {
                Response<String> resp = super.parseNetworkResponse(response);

                return Response.success(resp.result, VolleyCache.parseIgnoreCacheHeaders(response, TimeUnit.DAYS.toMillis(30L)));
            }
        };
        request.setShouldCache(true);
        request.setRetryPolicy(new DefaultRetryPolicy(7000, DefaultRetryPolicy.DEFAULT_MAX_RETRIES, DefaultRetryPolicy.DEFAULT_BACKOFF_MULT));
        VolleySingleton.getInstance(this).addToRequestQueue(request);
    }

}

File: app/src/main/java/org/favcode54/views/BareWebActivity.kt
package org.favcode54.views

import android.annotation.SuppressLint
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.view.View
import android.webkit.WebSettings
import com.android.volley.DefaultRetryPolicy
import com.android.volley.NetworkResponse
import com.android.volley.Response
import com.android.volley.toolbox.StringRequest
import com.jadebyte.naijaloaded.network.VolleyCache
import com.jadebyte.naijaloaded.network.VolleySingleton
import kotlinx.android.synthetic.main.activity_bare_web.*
import kotlinx.android.synthetic.main.content_bare_web.*
import org.favcode54.BaseActivity
import org.favcode54.MyWebViewClient
import org.favcode54.R
import org.favcode54.network.MyVolleyError
import timber.log.Timber
import java.util.concurrent.TimeUnit

class BareWebActivity : BaseActivity() {

    companion object {
        const val URL = "PAGE_URL"
        const val TITLE = "PAGE_TITLE"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bare_web)
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.title = intent.getStringExtra(TITLE)
        applyFont(rootView)
        tintProgressBar(progressBar)

        val url = intent.getStringExtra(URL)
        Timber.i("onCreate: %s", url)
        initWebView()
        loadPageHtml(url)
    }


    @SuppressLint("SetJavaScriptEnabled", "AddJavascriptInterface")
    private fun initWebView() {
        val webSettings = webView.settings
        webSettings.javaScriptEnabled = true
        webView.setBackgroundColor(Color.TRANSPARENT)
        webView.webViewClient = MyWebViewClient(this)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            webSettings.layoutAlgorithm = WebSettings.LayoutAlgorithm.TEXT_AUTOSIZING
        } else {
            webSettings.layoutAlgorithm = WebSettings.LayoutAlgorithm.NORMAL
        }
    }

    private fun loadPageHtml(url: String) {
        progressBar.visibility = View.VISIBLE
        val request = object : StringRequest(url,
                Response.Listener { response ->
                    progressBar.visibility = View.GONE
                    webView.loadDataWithBaseURL(url, response, "text/html", "UTF-8", null)
                },
                Response.ErrorListener { error ->
                    Snackbar.make(rootView, MyVolleyError.errorMessage(error, this)!!, Snackbar.LENGTH_LONG).show()
                    finish()
                })
        {
            override fun parseNetworkResponse(response: NetworkResponse?): Response<String> {
                val resp = super.parseNetworkResponse(response)
                return Response.success(resp.result, VolleyCache.parseIgnoreCacheHeaders(response!!, TimeUnit.DAYS.toMillis(30))) // Cache
                // for 30 days
            }
        }

        request.setShouldCache(true)
        request.retryPolicy = DefaultRetryPolicy(7000, DefaultRetryPolicy.DEFAULT_MAX_RETRIES, DefaultRetryPolicy.DEFAULT_BACKOFF_MULT)
        VolleySingleton.getInstance(this).addToRequestQueue(request)
    }

}
